SS14.Launcher/Models/Data/DataManager.cs

```diff
@@ -535,13 +553,54 @@ public bool Remove(ServerFilter item)
         }
 
         public bool Contains(ServerFilter item) => _parent._filters.Contains(item);
+        public void CopyTo(ServerFilter[] array, int arrayIndex) => _parent._filters.CopyTo(array, arrayIndex);
+        public int Count => _parent._filters.Count;
+        public bool IsReadOnly => false;
+    }
+
+    private sealed class HubCollection : ICollection<Hub>
+    {
+        private readonly DataManager _parent;
 
-        public void CopyTo(ServerFilter[] array, int arrayIndex)
+        public HubCollection(DataManager parent)
         {
-            _parent._filters.CopyTo(array, arrayIndex);
+            _parent = parent;
         }
 
-        public int Count => _parent._filters.Count;
+        public IEnumerator<Hub> GetEnumerator() => _parent._hubs.GetEnumerator();
+        IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+
+        public void Add(Hub item)
+        {
+            _parent._hubs.Add(item);
+
+            _parent.AddDbCommand(cmd => cmd.Execute(
+            "INSERT INTO Hubs (Address, Priority) VALUES (@Address, @Priority)",
+            new { item.Address, item.Priority }));
+        }
+
+        public void Clear()
+        {
+            _parent._hubs.Clear();
+
+            _parent.AddDbCommand(cmd => cmd.Execute("DELETE FROM Hubs"));
+        }
+
+        public bool Remove(Hub item)
+        {
+            if (!_parent._hubs.Remove(item))
+                return false;
+
+            _parent.AddDbCommand(cmd => cmd.Execute(
+                "DELETE FROM Hubs WHERE Address = @Address AND Priority = @Priority",
```

Comment:
Address is the primary key, you shouldn't be needing to filter for `Priority` in the `WHERE`.